src/Avatar.ts

```diff
@@ -112,6 +113,31 @@ export function defaultAvatarUrlForString(s: string): string {
     return dataUrl;
 }
 
+/**
+ * If client is particular about role colors  user role/user directory category then
+ * approach below would be sufficient to replace colors configured in custom config as well as color
+ * You can configure your own color in object in config file
+ * Not having this config or not having color in config parameter would default
+ * avatar colors back to matrix color
+ * @param {string} s
+ * @param {string} randomlySelectedAvatarColor randomly selected color from config
+ * @param {object} configAvatarColors needs to be obj{key:value} pair type in config
+ * @param {object} avatarColors needs to be passed from config and read via config.ts file
+*/
+function getSpecificColorFromConfig(s: string): string {
+    let configAvatarColors = config.avatarColors;
+    let avatarColors = configAvatarColors[s];
+    let numOfColors = Object.keys(configAvatarColors).length;
+    let total = 0;
+    for (let i = 0; i < s.length; ++i) {
+        total += s.charCodeAt(i);
+    }
+    const colorIndex = total % numOfColors;
+    const defaultToOtherColors = String(Object.values(configAvatarColors)[colorIndex]);
+    let randomlySelectedAvatarColor = String(Object.values(configAvatarColors)[Math.floor(Math.random() * numOfColors)]);
```

Comment:
It captures both nullish and undefined results and goes to next one but yeah chance of randomlySelected color being used are based on error in calculating text. But it is a rare case of 5-10% errored cases of name not being passed in function when. These may include:
1) System failure where name is not string or name is passed as null into this calculator.
2)  Name is not in valid text format 
3) Developer changing something erratically on function that sends name into this function as parameter, then it cant resolve name as string so returns undefind/null, and in that case use randomly selected color instead of using matrix green colors.
These would result final way of going to last one as randomly selected color.
**Please find my new changes pushed for requested changes as well for previous comments in new commits**